src/core/services/TransactionService.ts

```diff
@@ -101,27 +101,36 @@ export class TransactionServiceImpl implements TransactionService {
   public handleTransaction = async ({
     network,
     tx,
+    useExternalService = false,
     renderNotification = true,
   }: HandleTransactionProps): Promise<TransactionReceipt> => {
     const { NETWORK_SETTINGS } = getConfig();
     const currentNetworkSettings = NETWORK_SETTINGS[network];
+    const useNotifyService = useExternalService && currentNetworkSettings.notifyEnabled;
+
     let updateNotification: UpdateNotification | undefined;
     let dismissNotification: () => void = () => undefined;
-    try {
-      if (renderNotification) {
-        if (currentNetworkSettings.notifyEnabled) {
-          notify.hash(tx.hash);
-        } else {
-          const { update, dismiss } = notify.notification({
-            eventCode: 'txSentCustom',
-            type: 'pending',
-            message: 'Your transaction has been sent to the network',
-          });
-          updateNotification = update;
-          dismissNotification = dismiss;
-        }
+    let notifyServiceFailed = false;
+    if (renderNotification && useNotifyService) {
+      try {
+        notify.hash(tx.hash);
+      } catch (error: any) {
+        notifyServiceFailed = true;
+        console.error(error);
       }
+    }
+
+    if (renderNotification && (!useNotifyService || notifyServiceFailed)) {
```

Comment:
I'm not sure I follow this, so we are checking `!useNotifyService || notifyServiceFailed` and then using `notify.notification()`? Isn't that using the notify service?

Reply:
`notify.notification()` is meant for custom messages, more as a UI component, and should not be utilizing Blocknative service AFAIK. Remember reading that somewhere but cant find it now https://docs.blocknative.com/notify#notification.

We actually do this for fantom, since its disabled by default https://github.com/yearn/yearn-finance-v3/blob/develop/src/config/constants/index.ts#L67

Reply:
Ah I see, so what we're checking with `notifyServiceFailed` is if `blockNativeNotifyServiceFailed` -- if so, I think it's worth to rename as it's a bit confusing. Initially I was wondering why we are using the "notify" service by calling `notification()` on it if it failed for example

Reply:
yeah, probably makes sence to refactor to better naming. will do

Reply:
changed here https://github.com/yearn/yearn-finance-v3/pull/674/commits/e4da2120c74fded35f559f4fe787f85e84f99719

Reply:
I was also thinking we should refactor the `notifyServiceFailed` to `blockNativeNotifyServiceFailed`

Reply:
noticed it too and edited my comment after :P